features/admin.actions.v1/api/use-get-action-types.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "@wso2is/admin.core.v1/hooks/use-request";
+import { store } from "@wso2is/admin.core.v1/store";
+import { HttpMethods } from "@wso2is/core/models";
+import { ActionsConstants } from "../constants/actions-constants";
+import { ActionTypesResponseInterface } from "../models/actions";
+
+/**
+ * Hook to get the action types.
+ *
+ * @param shouldFetch - Should fetch the data.
+ * @returns SWR response object containing the data, error, isLoading, isValidating, mutate.
+ */
+const useGetActionTypes = <Data = ActionTypesResponseInterface[], Error = RequestErrorInterface>
+    (shouldFetch: boolean = true): RequestResultInterface<Data, Error> => {
+
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: `${ store.getState().config.endpoints.actions }/${ ActionsConstants.TYPES_DIR }`
+    };
+
+    const { data, error, isLoading, isValidating, mutate } = useRequest<Data, Error>
+    (shouldFetch ? requestConfig : null, { shouldRetryOnError: false });
```

Comment:
Formatting improvement.
```suggestion
    const { data, error, isLoading, isValidating, mutate } = useRequest<Data, Error>(
        shouldFetch ? requestConfig : null,
        { shouldRetryOnError: false }
    );
```

Reply:
Please check other APIs as well.

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/5e3564e0797c09f4f20b71837b928bc9fa57c847